airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java

```diff
@@ -168,10 +173,21 @@ protected AutoCloseableIterator<JsonNode> queryTableFullRefresh(final BigQueryDa
                                                                   final SyncMode syncMode,
                                                                   final Optional<String> cursorField) {
     LOGGER.info("Queueing query for table: {}", tableName);
-    return queryTable(database, String.format("SELECT %s FROM %s",
-        enquoteIdentifierList(columnNames, getQuoteString()),
-        getFullyQualifiedTableNameWithQuoting(schemaName, tableName, getQuoteString())),
-        tableName, schemaName);
+     // This corresponds to the initial sync for in INCREMENTAL_MODE, where the ordering of the records matters
+    // as intermediate state messages are emitted (if the connector emits intermediate state).
+    if (syncMode.equals(SyncMode.INCREMENTAL) && getStateEmissionFrequency() > 0) {
```

Comment:
getStateEmissionFrequency will always be > 0?

Reply:
Yes right now I have hard-coded it to 10000

Reply:
Not needed